docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,116 @@
+| Device Commands | array | Situational | When the device command is not associated to any reading or event, the device commands list can be modified (i.e., adding or removing the associated device commands).  Once the device command is associated to a reading or event, only new commands can be added to the array.  Deleting is prohibited.  Device commands are optional and so the array can always be empty. |
+| *Device Resource Properties* |||
+| Description | string | Anytime | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Name | string | Situational | When the device profile is not associated to any reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device resource to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided. |
+| Tag | string | Situational | As this property does not affect any functionality, this property can be changed at any time – to include setting the string to an empty string. |
+| Properties | Resource Properties | Situational | When the device resource (directly or indirectly through device command) is not associated to any reading or event, this field can be modified.  Because the ValueType and ReadWrite properties of this field are required, it can never be empty. |
+| Attributes | map | Situational | When the device resource (directly or indirectly through device command) is not associated to any reading or event, this field can be modified.|
+| *Device Command Properties* ||||
+| Name | string | Situational | When the device command is not associated to any reading or event, this field can be modified.  It can never be blank and must be unique across the entire profile. |
+| IsHidden | bool | Anytime | This field allows the device command to be exposed (default) via the command service.  This change does not affect the behavior but only visibility.  This field is optional but is false by default when not provided.|
+| ReadWrite | string | Situational | When the device command is not associated to any reading or event, this field can be modified.  This field is required.  As the field is expected to be from an enumeration of R, W and RW, it can never be empty. |
+| ResourceOperations | array | Situational | When the device command is not associated to any reading or event, this field can be modified.  This field is required and can never be empty. |
```

Comment:
Same as previous

Reply:
But do we really want restrict this? What if I have any existing device command called `ReadAll`, and I add a new device resource. Shouldn't I be able to extend my command?